### DISCUSSION TITLE
June 2024 Hackathon Moeschberg

### DISCUSSION THREAD

**Discussion by cmutel** [Depart-de-Sentier/sentier.dev#23] [Announcements]
# Sentier.dev June 2024 Hackathon Moeschberg

## Logistics

* What: Mini-hackathon on glossary data schema and storage, and fleshing out model writing and orchestration
* Where: [Möschberg seminar hotel](https://www.moeschberg.ch/), [Möschberg 7b, 3506 Grosshöchstetten, Switzerland](https://search.ch/map/Grossh%C3%B6chstetten,M%C3%B6schberg-7b)
* When: 9:30 June 26 - 16:00 June 27

## Concept

The hackathon builds on the discussions from our [first workshop](https://github.com/sentier-dev/sentier.dev/discussions/1), and aims to make the concepts more concrete, as well as getting some (proto-)code down on virtual paper so that we can think about iterating on products instead of talking about generalities.

The DdS Sentier.dev platform has three basic components:

1. A glossary - shared set of terms which forms better identifiers for produces, processes, units, locations, and models, including relationships across existing taxonomies.
2. Model infrastructure, including base classes, tabular data storage, and orchestration
3. Meta-analysis of the system, including quality assessment, prioritization of data quality or quantity improvements, and use of innovative data sources to refine and validate our local or global totals.

![DdS concept overview](https://github.com/Depart-de-Sentier/sentier.dev/assets/645217/daf2b8a7-1164-49fd-a9b2-5b33f4cc2927)

## Specific example: Parameterized wind turbine construction and operation

[Model link](https://github.com/romainsacchi/LCA_WIND_DK/tree/master), [Web UI](https://viewer.webservice-energy.org/lca-wind-dk/)

Wind turbines turn fusion energy from the sun into electrical energy via kinetic energy in the wind as a result of temperature differentials in the atmosphere. They have major components, such as the foundation (usually concrete), tower (usually steel or concrete), nacelle (oval thing on top of the tower, has gearbox, generator, electrical equipment, lubrication, etc), and turbine blades (usually fiberglass). There are scaling laws for the amount of materials needed, as larger turbines are higher, heavier, and therefore require substantially more material.

A **wind turbine production model** would take in the following input parameters:

* Maximum power (kW or mW)
* Height (could be ~f(maximum power))
* Generator type (with or without gearbox)
* Number of blades (default is 3)
* Onshore or offshore
* Brand (default is average production)

The outputs would be:

* *Maximum power (kW or mW)*
* *Height (could be ~f(maximum power))*
* *Generator type (with or without gearbox)*
* *Number of blades (default is 3)*
* *Onshore or offshore*
* *Brand (default is average production)*
* Materials and energy needed for construction of a specific turbine
* A [wind turbine power curve](https://en.wikipedia.org/wiki/Wind_turbine_design)
* Expected lifetime

A **wind turbine operation model** would take in the following input parameters:

* **Location** (gives onshore versus offshore)
* A given maintenance regime (default is global average)
* Turbine layout if in a larger set of turbines (optional)
* Maximum power
* Height
* Generator type (with or without gearbox)
* Number of blades
* A [wind turbine power curve](https://en.wikipedia.org/wiki/Wind_turbine_design)
* Expected lifetime

The outputs would be:

* Annual electricity production
* Materials and energy needed for operation and maintenance

### Adding Models to the Glossary
- #24
- #25
- #26 

### Prioritisation Dashboard
- #27  

### Using the glossary to answer questions
- #28

### Discussion on current glossary data schema

**Comment by mklarmann** [comment]
Example of a data-table:
<img width="1218" alt="Bildschirmfoto 2024-06-27 um 13 23 28" src="https://github.com/sentier-dev/sentier.dev/assets/328482/e7570ff9-f1c5-48f2-b0c2-bc626af6d151">

**Comment by cswh** [comment]
See
https://github.com/YannickSchubert/moeschberg_example_models
for the results of subgroup "conceptual prototype/model implementation/user stories".
